dsl/src/commonMain/kotlin/it/unibo/collektive/aggregate/api/Aggregate.kt

```diff
@@ -0,0 +1,66 @@
+package it.unibo.collektive.aggregate.api
+
+import it.unibo.collektive.ID
+import it.unibo.collektive.aggregate.api.YieldingContext.YieldingResult
+import it.unibo.collektive.field.Field
+
+/**
+ * Models the minimal set of aggregate operations.
+ * Holds the [localId] of the device executing the aggregate program.
+ */
+interface Aggregate {
+    /**
+     * The local [ID] of the device.
+     */
+    val localId: ID
+
+    /**
+     * This function computes the local value of e_i, substituting variable n with the nvalue w of
+     * messages received from neighbours, using the local value of e_i ([initial]) as a default.
+     * The exchange returns the neighbouring or local value v_r from the evaluation of e_r applied to the [body].
+     * e_s evaluates to a nvalue w_s consisting of local values to be sent to neighbour devices δ′,
+     * which will use their corresponding w_s(δ') as soon as they wake up and perform their next execution round.
+     *
+     * Often, expressions e_r and e_s coincide, so this function provides a shorthand for exchange(e_i, (n) => (e, e)).
+     *
+     * ## Example
+     * ```
+     * exchange(0) { f ->
+     *  f.mapField { _, v -> if (v % 2 == 0) v + 1 else v * 2 }
+     * }
+     * ```
+     * The result of the exchange function is a field with as messages a map with key the id of devices across the
+     * network and the result of the computation passed as relative local values.
+     */
+    fun <X> exchange(initial: X, body: (Field<X>) -> Field<X>): Field<X>
+
+    /**
+     * Same behavior of [exchange] but this function can yield a [Field] of [Ret] value.
+     */
+    fun <Init, Ret> exchanging(
+        initial: Init,
+        body: YieldingContext<Field<Init>, Field<Ret>>.(Field<Init>) -> YieldingResult<Field<Init>, Field<Ret>>,
+    ): Field<Ret>
```

Comment:
May I suggest to keep the same syntax for the generic types? 
for example in the `repeating` function we use as generics `<Initial, Return>`, maybe it's better like this:
```suggestion
    fun <Initial, Return> exchanging(
        initial: Initial,
        body: YieldingContext<Field<Initial>, Field<Return>>.(Field<Initial>) -> YieldingResult<Field<Initial>, Field<Return>>,
    ): Field<Return>
```

or eventually change the others into `<Init, Ret>`

Reply:
Using <Initial, Return> we exceed 120 cols, and I have not found a good breaking to reduce within 120 cols.

Reply:
Solved in 2352a449d97920d1470ebd62c44c7f76e7b60b6f